File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
        Stack<Character> stack = new Stack<>();
        
        for (int i = 0; i < s.length(); i++) {
            char current = s.charAt(i);
            
            if (current == '(') {
                stack.push(current);
            } else if (current == ')' && stack.isEmpty()) {
                return false;
            } else {
                stack.pop();
            }
        }
    
        return stack.isEmpty();
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):    
    bracket_stack = []
    
    for i in s:
        if i == "(":
            bracket_stack.append(i)
        
        elif i == ")":
            if len(bracket_stack) != 0:
                bracket_stack.pop()
            else:
                return False
            
    return len(bracket_stack) == 0